src/Producer/SyncProcess.php

```diff
@@ -151,18 +153,6 @@ protected function convertMessage(array $data): array
         $topicInfos = $broker->getTopics();
 
         foreach ($data as $value) {
-            if (! isset($value['topic']) || ! \trim($value['topic'])) {
```

Comment:
I think we should rather do:

```php
foreach ($recordSet as $record) { // just renaming the variables for consistency
    $this->ensureRecordValidity($record);

    // ...
}
```

This solves the double iteration issue mentioned before